app/config_test.go

```diff
@@ -4,17 +4,54 @@
 package app
 
 import (
+	"io"
+	"io/ioutil"
+	"os"
 	"strconv"
+	"strings"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/store/sqlstore"
 	"github.com/mattermost/mattermost-server/utils"
 )
 
+func TestLoadConfig(t *testing.T) {
+	permConfig, err := os.Open(utils.FindConfigFile("config.json"))
```

Comment:
@hanzei, can you decouple this from the existing `config.json` file and just write out something directly to a temporary file instead? I think it would simplify the testing code from having to find the existing one, copy it, and re-read the file back into memory just to modify a line of interest.